src/bin/tree-sitter-graph/main.rs

```diff
@@ -138,17 +138,13 @@ fn main() -> Result<()> {
 
     let functions = Functions::stdlib();
     let mut config = ExecutionConfig::new(&functions, &globals_).lazy(lazy);
-    let graph = file
-        .execute(
-            &tree,
-            source_path,
-            &source,
-            tsg_path,
-            &tsg,
-            &mut config,
-            &NoCancellation,
-        )
-        .with_context(|| format!("Cannot execute TSG file {}", tsg_path.display()))?;
+    let graph = match file.execute(&tree, &source, &mut config, &NoCancellation) {
+        Ok(graph) => graph,
+        Err(e) => {
+            eprint!("{}", e.display_pretty(source_path, &source, tsg_path, &tsg));
```

Comment:
Generally it's a code smell to `fprintf` from a library like this.  Is this message intended to eventually make it into Splunk?  Do we need to bite the bullet and pull in the `log` or `tracing` crate so that callers can control whether and how these messages are printed?  (And if so, I'm fine with that being a TODO comment here + a follow-on PR)

(If we do go down that route, `tracing` would be my preference, because then we could configure the log output to match what's happening on the Go side, and we'd end up with nicely structured log messages from both Go and Rust in the same place)

Reply:
This is the CLI code, not the library, so I think it should be okay here. For library code, I completely agree, though.

Reply:
Oh yes you're right!  I withdraw my objection :smile: